### PR TITLE
feat: add view skills by role

### PR DESCRIPTION
**Pull request by bayberryy**
feat: add view-skills-by-role page

### CODE REVIEW

File: client/pages/view-skills-by-role.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                <div className="body">
+
+                    {skillList && skillList?.length == 0 &&(
+                        <div className="notFound">No Skills Found</div>
+                    )}
+                {skillList && 
+                    skillList?.length>0 && 
+                    skillList?.map(role =>{
+                    return(
+                        <div className="body__item">
```
Comment (regexyl) [inline client/pages/view-skills-by-role.tsx]:
```suggestion
                        <div className="body__item" key={role}>
```
This should fix the error :D (you can view the error by running `tsc --noEmit` under the client folder). Every list-rendered component would need a key prop for React to identify it for maintaining its sequence. (ref: [this](https://reactjs.org/docs/lists-and-keys.html)).

File: client/pages/view-skills-by-role.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    >(skills);
+    const [text, setText] = React.useState<string>('');
+
+    const handleOnClick = () => {
+        const findSkills = 
+            skillList && skillList?.length > 0 
+            ? skillList?.filter((s) => s?.role == text)
+            : undefined;
+
+            console.log(findSkills);
```
Comment (regexyl) [inline client/pages/view-skills-by-role.tsx]:
```suggestion
```

File: client/pages/view-skills-by-role.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                            <Card.Content>
+                                <Text b my={0}>Role: {role?.role}</Text>
+                                </Card.Content>
+                                <Divider h="1px" my={0} />
+                                <Card.Content>
+                                <Text>{role?.label}</Text>
+                                <Text>{role?.description}</Text>
+                            {/* <p>Role: {role?.role}</p>
+                            <p>Skill1: {role?.label}</p>
+                            <p>Skill: {role?.description}</p> */}
```
Comment (regexyl) [inline client/pages/view-skills-by-role.tsx]:
```suggestion
```

**Review by regexyl — CHANGES_REQUESTED** [review]
Yay nice!! Especially with the use of types and React hooks 😄 

I think we'll eventually migrate all these to the `view-skils.tsx` page so users don't have to navigate to a separate page to filter by role, but we can clean up this and push it first.

File: client/pages/view-skills-by-role.tsx (hunk 1 above)
Comment (regexyl) [reply client/pages/view-skills-by-role.tsx]: Here are the error logs that show why the build is failing:

`62:25  Error: Missing "key" prop for element in iterator  react/jsx-key`
image

This error doesn't prevent the webpage from rendering, it's just that there may be some behavioral issues when we interact with the components (which isn't very obvious on first glance).

Comment (bayberryy) [reply client/pages/view-skills-by-role.tsx]: If I add `key={role}` , I'll have error on the code
image

Comment (regexyl) [reply client/pages/view-skills-by-role.tsx]: What's the error shown when you hover on `key`?

Comment (bayberryy) [reply client/pages/view-skills-by-role.tsx]: There is no error in terminal and website works fine. Its just that there the `key` is underlined. I just push it.

File: client/pages/view-skills-by-role.tsx (hunk 2 above)
Comment (bayberryy) [reply client/pages/view-skills-by-role.tsx]: Removed 👍

File: client/pages/view-skills-by-role.tsx (hunk 3 above)
Comment (bayberryy) [reply client/pages/view-skills-by-role.tsx]: Removed 👍

**Pull request closed without merging**